FabiWare/FabiWare.ino

```diff
@@ -344,18 +353,34 @@ void loop() {
 
 void handlePress (int buttonIndex)   // a button was pressed
 {
+  static uint32_t doublePressTimestamp =0;
+  #ifdef DEBUG_OUTPUT
+    Serial.print("press button "); Serial.print(buttonIndex);
+  #endif
+
+  if (settings.dp > 0) {     // check if double press condition met
+      if ((millis() - doublePressTimestamp) < settings.dp) {
+        // Serial.println("skip to next Slot!");
+        performCommand(CMD_NE,0,0,0);  // activate next slot if yes!
```

Comment:
Yes - currently there is only the standard action "next slot" for the double press activity. Per default, this function is disabled (-> should not create unexpected behaviour).

Allowing free definition of a double-press action for every button would need lots of efforts (both in the firmware and in the GUIs). And it would dramatically increase RAM consumption due to the inefficient memory organisation).

Adding the single "next slot" function is IMO a good compromise - it is primarily important for single-switch operation and opens up many new possibilities for this use case.

It is a good idea to return after the slot change (and not perfoming the function afterwards) - it will add this to the code!

Reply:
yes, I know it would be much more effort for free double-press actions for each button, we don't have to implement it now.
In AsTeRICS Grid I wait for the timeout before doing the single-press action in order to prevent doing both the single-press action and the double-press action. This maybe would also be an option here, but we also don't have to implement it now.

Reply:
i thought about this - but especially for games you want an immediate reaction for the single-press action, waiting for the double press timeout would create a latency here ....

also the immediate return after the slot change caused unwanted behaviour for the single-switch mouse control use case (because there you want the mouse movement to happen immediately after the double press).

so i leave everything as it is for now .... we can discuss changes as needed for the most useful use cases.
i attach a demo setting for 1-switch mouse control here (pls rename .txt to .set):

[mousemove_test.txt](https://github.com/asterics/FABI/files/6384013/mousemove_test.txt)

Reply:
ok, makes sense. so many possibilities and use cases...